sap-spring-properties/src/test/java/com/sap/cloud/environment/servicebinding/SapSpringPropertiesServiceBindingAccessorTest.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    @Test
+    void getBindingsInvalidEmptyName()
+    {
+        final MockEnvironment mockEnvironment = TestResource.getInvalidPropertiesEmptyName();
+        final SapServiceBindingEnvironmentPostProcessor sapServiceBindingEnvironmentPostProcessor =
+            new SapServiceBindingEnvironmentPostProcessor();
+        sapServiceBindingEnvironmentPostProcessor.postProcessEnvironment(mockEnvironment, null);
+
+        SapSpringPropertiesServiceBindingAccessor cut = new SapSpringPropertiesServiceBindingAccessor();
+
+        assertTrue(cut.getServiceBindings().isEmpty());
+    }
+
+    @Test
+    void getBindingsInvalidNoCredentials()
+    {
+        final MockEnvironment mockEnvironment = TestResource.getInvalidPropertiesNoCredentials();
+        final SapServiceBindingEnvironmentPostProcessor sapServiceBindingEnvironmentPostProcessor =
+            new SapServiceBindingEnvironmentPostProcessor();
+        sapServiceBindingEnvironmentPostProcessor.postProcessEnvironment(mockEnvironment, null);
+
+        SapSpringPropertiesServiceBindingAccessor cut = new SapSpringPropertiesServiceBindingAccessor();
+
+        assertTrue(cut.getServiceBindings().isEmpty());
+    }
+
+    private void assertContainsXsuaaBinding( List<ServiceBinding> serviceBindings )
+    {
+        Optional<ServiceBinding> maybeXsuaaBinding =
+            serviceBindings
+                .stream()
+                .filter(
+                    binding -> binding.getServiceName().isPresent() && binding.getServiceName().get().equals("xsuaa"))
+                .findFirst();
+        assertTrue(maybeXsuaaBinding.isPresent());
+        ServiceBinding xsuaaCertBinding = maybeXsuaaBinding.get();
+        assertTrue(xsuaaCertBinding.getServicePlan().isPresent());
+        assertEquals("broker", xsuaaCertBinding.getServicePlan().get());
+        assertTrue(xsuaaCertBinding.getServiceName().isPresent());
+        assertEquals("xsuaa", xsuaaCertBinding.getServiceName().get());
+        assertEquals("xsuaa-test", xsuaaCertBinding.getName().get());
+        assertEquals("https://localhost:8080", xsuaaCertBinding.getCredentials().get("certurl"));
+    }
+
+    private void assertContainsServiceManagerBindings( List<ServiceBinding> serviceBindings )
+    {
+        Map<String, ServiceBinding> serviceManagerBindings =
+            serviceBindings
+                .stream()
+                .filter(
+                    binding -> binding.getServiceName().isPresent()
+                        && binding.getServiceName().get().equals("service-manager"))
+                .collect(
+                    Collectors
+                        .toMap(serviceBinding -> serviceBinding.getName().get(), serviceBinding -> serviceBinding));
+        assertEquals(2, serviceManagerBindings.size());
+
+        ServiceBinding serviceManagerBinding1 = serviceManagerBindings.get("service-manager-test");
+        assertEquals("service-manager", serviceManagerBinding1.getServiceName().get());
```

Comment:
(Comment/Nitpick)

No action required. I just want to advertise `org.assertj.core.api.Assertions.assertThat` API as we use in other test classes. _AssertJ_ has powerful convenience methods that can be used for asserting. Great tool.

Reply:
will have a look. thank you for the hint!